bin/update-ts-project-references.ts

```diff
@@ -0,0 +1,133 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+
+import cp from "child_process";
+import fs from "fs";
+import fsP from "fs/promises";
+import path from "path";
+import { getDependentsGraph } from "@changesets/get-dependents-graph";
+import { getPackages } from "@manypkg/get-packages";
+
+type GetPackagesResult = Awaited<ReturnType<typeof getPackages>>;
+type Package = GetPackagesResult["packages"][number];
+type DependencyGraph = Map<string, Set<string>>;
+type TSConfig = { references?: { path: string }[] };
+
+const stringArrayEquals = (arrA: string[], arrB: string[]): boolean => {
+  return (
+    arrA.length === arrB.length && arrA.every((itemA, i) => itemA === arrB[i])
+  );
+};
+
+let prettier: typeof import("prettier") | undefined;
+
+const format = (filePath: string, content: string): string => {
+  if (!prettier) {
+    prettier = require("prettier") as typeof import("prettier");
+  }
+  const config = prettier.resolveConfig(filePath);
+  return prettier.format(content, {
+    ...config,
+    filepath: filePath,
+  });
+};
+
+const getDependencyGraph = (
+  getPackagesResult: GetPackagesResult
+): DependencyGraph => {
+  const graph: DependencyGraph = new Map();
+
+  for (const [dependency, dependents] of getDependentsGraph(
+    getPackagesResult
+  )) {
+    for (const dependent of dependents) {
+      let dependencies = graph.get(dependent);
+      if (!dependencies) {
+        dependencies = new Set();
+        graph.set(dependent, dependencies);
+      }
+      dependencies.add(dependency);
+    }
+  }
+
+  return graph;
+};
+
+const updateFile = async (filePath: string, content: string) => {
+  await fsP.writeFile(filePath, format(filePath, content));
+  cp.spawnSync("git", ["add", filePath]);
+};
+
+const updateTSConfig = async (
```

Comment:
nit: we are using this convention
```suggestion
const updateTsConfig = async (
```

Reply:
Ok, I've made this change and also adjusted the `TSConfig` type to `TsConfig`